table/block_based/filter_policy.cc

```diff
@@ -338,7 +338,8 @@ FilterBitsReader* BloomFilterPolicy::GetFilterBitsReader(
     return new AlwaysFalseFilter();
   }
 
-  char raw_num_probes = contents.data()[len_with_meta - 5];
+  int8_t raw_num_probes =
+      static_cast<int8_t>(contents.data()[len_with_meta - 5]);
```

Comment:
Maybe it's obvious to someone else but I didn't quite get why `char` fails the assertion but int8_t doesn't. Can you explain a little bit?

Reply:
On ARM, char is effectively uint8_t, so can be > 127

Reply:
But the source data couldn't have the first bit set?

If it is > 127, then on x86, it is < 1 so the other assertion would hit too?